src/courtroom.cpp

```diff
@@ -773,26 +796,20 @@ void Courtroom::handle_chatmessage(QStringList p_contents)
   // reset our ui state if client just spoke
   if (m_cid == f_char_id && is_system_speaking == false)
   {
-    ui_ic_chat_message->clear();
-
-    // reset states
-    ui_pre->setChecked(ao_config->always_pre_enabled());
-    list_sfx();
-    ui_sfx_list->setCurrentItem(
-        ui_sfx_list->item(0)); // prevents undefined errors
-
-    m_shout_state = 0;
-    draw_shout_buttons();
-
-    m_effect_state = 0;
-    draw_effect_buttons();
-
-    m_wtce_current = 0;
-    draw_judge_wtce_buttons();
+    // If the server does not have the feature of acknowledging our MS
+    // messages, assume in this if that the message is proof the server
+    // acknowledged our message. It is not quite the same, as it is
+    // possible the server crafted a message with the same char_id
+    // as the client, but the client did not send that message, but it is
+    // the best we can do.
+    if (!ao_app->ackMS_enabled)
+      handle_acknowledged_ms();
```

Comment:
People are expected to be using both the proper client and the corresponding matching version of the server to go along with it. This is not needed.